pyk/src/pyk/ktool/kprint.py

```diff
@@ -0,0 +1,23 @@
+from pathlib import Path
+
+from ..kast import buildSymbolTable, prettyPrintKast, readKastTerm
+from ..utils import hash_str
+
+
+class KPrint:
+    """Given a kompiled directory, build an unparser for it.
+    """
+
+    def __init__(self, kompiledDirectory):
+        self.kompiledDirectory = Path(kompiledDirectory)
+        self.definition = readKastTerm(self.kompiledDirectory / 'compiled.json')
+        self.symbolTable = buildSymbolTable(self.definition, opinionated=True)
+        self.definitionHash = hash_str(self.definition)
+
+    def prettyPrint(self, kast, debug=False):
+        """Given a KAST term, pretty-print it using the current definition.
+
+        -   Input: KAST term in JSON.
+        -   Output: Best-effort pretty-printed representation of the KAST term.
+        """
+        return prettyPrintKast(kast, self.symbolTable, debug=debug)
```

Comment:
Maybe we should move the definition of `prettyPrintKast` into this file?

Reply:
And encourage people to go thorugh the class interface.

Reply:
I wouldn't do that. Having the ability to quickly print a simple `KAst` based on a custom symbol table can be very useful. But `KPrint` requires a kompiled definition, so that makes this very cumbersome.

Reply:
Well, I wasn't thinking we would move the definition of prettyPrintKast into the class itself, just move the definition of that function here. But not sure if it's better. It just seems like a logical organization of the code, but no strong opinion on it.

Reply:
Oh yeah, that makes sense, sorry for the confusion.